fMRISurface/scripts/CreateDenseTimeseries.sh

```diff
@@ -15,8 +15,28 @@ TR_vol=`fslval "$NameOffMRI" pixdim4 | cut -d " " -f 1`
 
 #Some way faster and more concise code:
 
-${CARET7DIR}/wb_command -cifti-create-dense-timeseries "$OutputAtlasDenseTimeseries".dtseries.nii -volume "$NameOffMRI"_AtlasSubcortical_s"$SmoothingFWHM".nii.gz "$ROIFolder"/Atlas_ROIs."$GrayordinatesResolution".nii.gz -left-metric "$NameOffMRI"_s"$SmoothingFWHM".atlasroi.L."$LowResMesh"k_fs_LR.func.gii -roi-left "$DownSampleFolder"/"$Subject".L.atlasroi."$LowResMesh"k_fs_LR.shape.gii -right-metric "$NameOffMRI"_s"$SmoothingFWHM".atlasroi.R."$LowResMesh"k_fs_LR.func.gii -roi-right "$DownSampleFolder"/"$Subject".R.atlasroi."$LowResMesh"k_fs_LR.shape.gii -timestep "$TR_vol"
+${CARET7DIR}/wb_command -cifti-create-dense-timeseries \
+			"$OutputAtlasDenseTimeseries".dtseries.nii \
+			-volume "$NameOffMRI"_AtlasSubcortical_s"$SmoothingFWHM".nii.gz "$ROIFolder"/Atlas_ROIs."$GrayordinatesResolution".nii.gz \
+			-left-metric "$NameOffMRI"_s"$SmoothingFWHM".atlasroi.L."$LowResMesh"k_fs_LR.func.gii -roi-left "$DownSampleFolder"/"$Subject".L.atlasroi."$LowResMesh"k_fs_LR.shape.gii \
+			-right-metric "$NameOffMRI"_s"$SmoothingFWHM".atlasroi.R."$LowResMesh"k_fs_LR.func.gii -roi-right "$DownSampleFolder"/"$Subject".R.atlasroi."$LowResMesh"k_fs_LR.shape.gii \
+			-timestep "$TR_vol"
 
+#Assess for zeros in the final dtseries (e.g., due to incomplete spatial coverage)
+# (Note that earlier steps do an appreciable amount of dilation to eliminate zeros,
+# so zeros remaining in the CIFTI at this point represent a non-trivial issue with spatial coverage)
+${CARET7DIR}/wb_command -cifti-reduce "$OutputAtlasDenseTimeseries".dtseries.nii STDEV "$OutputAtlasDenseTimeseries".stdev.dscalar.nii
+Nnonzero=`${CARET7DIR}/wb_command -cifti-stats "$OutputAtlasDenseTimeseries".stdev.dscalar.nii -reduce COUNT_NONZERO`
+Ngrayordinates=`${CARET7DIR}/wb_command -file-information "$OutputAtlasDenseTimeseries".stdev.dscalar.nii | grep "Number of Rows" | awk '{print $4}'`
+PctCoverage=`echo "scale=4; 100 * ${Nnonzero} / ${Ngrayordinates}" | bc -l`
+echo "PctCoverage, Nnonzero, Ngrayordinates" >| "$OutputAtlasDenseTimeseries"_nonzero.stats.txt
+echo "${PctCoverage}, ${Nnonzero}, ${Ngrayordinates}" >> "$OutputAtlasDenseTimeseries"_nonzero.stats.txt
+# If we don't have full grayordinate coverage, save out a mask to identify those locations
+if [ "$Nnonzero" -ne "$Ngrayordinates" ]; then
```

Comment:
I'm not particularly fond of having an internal aspect of the data determine whether an output file gets generated.  If we added some automated fix for the zeros (more dilation), having the mask of zeros exist regardless of whether it is empty might make the code simpler.

Reply:
I did that in the case because due to the dilation, zeros in the CIFTI should be rare and the *_nonzero.stats.txt is written out regardless, so that provides the "evidence" that the check was performed on a particular set of processed data.  That said, if you and @glasserm prefer that we write out the *_nonzero.dscalar.nii file regardless, I can do that.

I don't see how adding more dilation impacts whether or not we should write out the map of nonzero grayordinates.  At some point, we can't just keep dilating, and frankly it seems like we already have a lot of dilation built in -- if I'm reading `RibbonVolumeToSurfaceMapping` correctly, we already dilate the "native" mesh data by 10 mm, and then dilate another 30 mm when we resample to the 32k_fsLR mesh.

Ideally, it would be great to have a dscalar that captures the grayordinates that contain "dilated" values themselves, but I don't know how easy that would be to implement...

Reply:
I prefer Mike's solution